build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -171,7 +171,9 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
         {
             var hostVersion = c.BuildContext.Get<HostVersion>("HostVersion");
             var configuration = c.BuildContext.Get<string>("Configuration");
-            string rid = c.BuildContext.Get<string>("TargetRID");
+            string rid = DotNetCli.Stage0.GetRuntimeId();
```

Comment:
>I would also suggest to rename the variable to "hostRID". And if you are doing this, I would suggest do this initialization in PrepareTargets.CommonInit and simply fetch it here from BuildContext.

I agree that those changes would be nice/better, but we should really do the absolute minimum code change to get this working as intended or we may have unintended breaks. The Core-Setup repo has a **lot** of hardcoded expectations that I don't want to test. 

I pushed a minimal fix that won't break the cross-arch build as long as it passes `TargetArch`